src/vs/platform/inputResults/browser/inputResultsService.ts

```diff
@@ -0,0 +1,94 @@
+
+import { Event } from 'vs/base/common/event';
+import { IDisposable, combinedDisposable, toDisposable } from 'vs/base/common/lifecycle';
+import { IContextKey, IContextKeyService, RawContextKey } from 'vs/platform/contextkey/common/contextkey';
+import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
+
+export const InputResultsViewResultsFocusedContextKey = new RawContextKey<boolean>('inputResultsViewResultsFocused', false);
+export const InputResultsViewInputFocusedContextKey = new RawContextKey<boolean>('inputResultsViewInputFocused', false);
+
+export const IInputResultsService = createDecorator<IInputResultsService>('inputResultsService');
+
+export interface IInputResultsView extends IRegisteredInputResultsView {
+	readonly onDidFocusInput: Event<any>;
+	readonly onDidBlurInput: Event<any>;
+	readonly onDidFocusResults: Event<any>;
+	readonly onDidBlurResults: Event<any>;
+	readonly onDispose: Event<any>;
+}
+
+export interface IRegisteredInputResultsView {
+	focusInput(): void;
+	focusResults(): void;
+}
+
+export interface IInputResultsService {
+	readonly _serviceBrand: undefined;
+
+	readonly lastFocusedView: IRegisteredInputResultsView | undefined;
+	register(view: IInputResultsView): IDisposable;
+}
+
+export class InputResultsService implements IInputResultsService {
+	declare readonly _serviceBrand: undefined;
+
+	private views: Set<IInputResultsView> = new Set();
+	private _lastFocusedView: IInputResultsView | undefined = undefined;
+	get lastFocusedView(): IInputResultsView | undefined { return this._lastFocusedView; }
+
+	private inputFocused: IContextKey<boolean>;
+	private resultsFocused: IContextKey<boolean>;
+
+	constructor(@IContextKeyService contextKeyService: IContextKeyService) {
+		this.inputFocused = InputResultsViewInputFocusedContextKey.bindTo(contextKeyService);
+		this.resultsFocused = InputResultsViewResultsFocusedContextKey.bindTo(contextKeyService);
+	}
+
+	private setLastFocusedView(view: IInputResultsView | undefined) {
+		this._lastFocusedView = view;
+	}
+
+	register(view: IInputResultsView): IDisposable {
+		if (this.views.has(view)) {
+			throw new Error('Cannot register the same view multiple times');
+		}
+
+		this.views.add(view);
+
+		return combinedDisposable(
+			view.onDidFocusInput(() => {
+				this.setLastFocusedView(view);
+				this.inputFocused.set(true);
+			}),
+			view.onDidBlurInput(() => {
+				if (!this.resultsFocused.get() && this._lastFocusedView === view) {
+					this.setLastFocusedView(undefined);
+				}
+				this.inputFocused.set(false);
+			}),
+			view.onDidFocusResults(() => {
+				this.setLastFocusedView(view);
+				this.resultsFocused.set(true);
+			}),
+			view.onDidBlurResults(() => {
+				if (!this.inputFocused.get() && this._lastFocusedView === view) {
+					this.setLastFocusedView(undefined);
+				}
+				this.resultsFocused.set(false);
+			}),
+			toDisposable(() => this.views.delete(view)),
+			view.onDispose(() => {
```

Comment:
I think that having this is redundant- when the view is disposed, it should dispose this disposable that you return here. So line 83 should do the same cleanup work that this handler does. That's the typical pattern we would use.

Reply:
Fix in commit https://github.com/microsoft/vscode/pull/187077/commits/1a084e61cdb0aa5f2a6bab172e3ba9c5abb64019